File: typescript/resistor-color-duo/resistor-color-duo.ts
export function decodedValue(colors: string[]): number {
  interface ColorsMapping {
    [propertyName: string]: string;
  }

  const ColorsValue: ColorsMapping = {
    black: "0",
    brown: "1",
    blue: "6",
    grey: "8",
    yellow: "4",
    violet: "7",
    orange: "3",
    green: "5"
  }
  
  return Number(`${ColorsValue[colors[0]] }${ColorsValue[colors[1]]}`)
}

File: python/resistor-color-duo/resistor_color_duo.py
RESISTORS_COLOR_DUO = {
    "black": '0',
    "brown": '1',
    "red": '2',
    "orange": '3',
    "yellow": '4',
    "green": '5',
    "blue": '6',
    "violet": '7',
    "grey": '8',
    "white": '9',
}

def value(colors):
    encoded = ''
    for color in colors:
        encoded += RESISTORS_COLOR_DUO[color]

    return int(encoded[0:2])

